src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -328,15 +335,15 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
                 services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionInvocationDispatcherShutdownManager>());
 
-                if (SystemEnvironment.Instance.IsRuntimeScaleMonitoringEnabled())
-                {
-                    services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
-                }
-                services.TryAddSingleton<FunctionsScaleManager>();
-
                 services.AddSingleton<IHostOptionsProvider, HostOptionsProvider>();
             });
 
+            if (SystemEnvironment.Instance.IsRuntimeScaleMonitoringEnabled())
+            {
+                // This is needed only to add ScaleMonitorService, all other services for scale are added by the time of the call
```

Comment:
If it's only ScaleMonitorService that we need to add, I wonder if we shouldn't just add it in all cases, and leverage the existing logic in the service that only starts the service when RTS is enabled?

Reply:
ConfigureWebJobsScale will be called from function host.